external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -266,26 +266,32 @@ private void doSeekRetriableTopicPartitions() {
             if (offsetAndMeta != null) {
                 kafkaConsumer.seek(rtp, offsetAndMeta.offset() + 1);  // seek to the next offset that is ready to commit in next commit cycle
             } else {
-                kafkaConsumer.seekToEnd(toArrayList(rtp));    // Seek to last committed offset
+                kafkaConsumer.seek(rtp, acked.get(rtp).committedOffset + 1);    // Seek to last committed offset
```

Comment:
Another PR is fine, but I don't think there needs to be a big change. If people are running with auto commit mode on they've already kind of decided that they don't care about at least once delivery (similar to disabling tuple acking), so I think simply making the spout ignore failed tuples should be fine. Alternatively auto commit mode could be removed from KafkaSpoutConfig and just be an optimization the spout enables if tuple acking is disabled. I don't think it makes sense to run with tuple acking on and auto commit on, nor does it really make sense to disable tuple acking and disable auto commit. @hmcl do you have an opinion on this?